src/RepairManager/Rules/ecc_rule.py

```diff
@@ -0,0 +1,95 @@
+from Rules.rules_abc import Rule
+from kubernetes import client, config
+import requests
+import json
+import os
+import time
+import yaml
+import util
+import logging
+
+def get_node_address_info():
+    config.load_kube_config()
+    api_instance = client.CoreV1Api()
+
+    service_account_list = api_instance.list_node()
+
+    # map InternalIP to Hostname
+    address_map = {}
+    
+    if (service_account_list):
+
+        for account in service_account_list.items:
+            internal_ip = None
+            hostname = None
+
+            for address in account.status.addresses:
+                if address.type == 'InternalIP':
+                    internal_ip = address.address
+
+                if address.type == 'Hostname':
+                    hostname = address.address
+            
+                address_map[internal_ip] = hostname
+
+    logging.debug('node address map: %s ' %  address_map)
+
+    return address_map
+
+
+
+def get_ECC_error_data(ecc_url):
+
+    response = requests.get(ecc_url)        
+    data = json.loads(response.text)
+    
+    if data:
+        ecc_metrics = data['data']['result']
+        logging.info('ECC error metrics from prometheus ' + json.dumps(ecc_metrics, indent=4))
+
+        return ecc_metrics
+
+
+
+class ECCRule(Rule):
+
+    def check_status(self):
+        try:
+            with open('rule-config.yaml', 'r') as rule_config:
+                config = yaml.safe_load(rule_config)
+                       
+            address_map = get_node_address_info()
+
+            ecc_url = config['ecc_prometheus_url']
+            ecc_metrics = get_ECC_error_data(ecc_url)
+    
+            global ecc_hostnames
+            ecc_hostnames = []
```

Comment:
Can this be a field of `ECCRule`? We can init `self.ecc_hostnames = []` in `def __init__(self)`. Try not to use global variables.